### ISSUE TITLE
CDC device receives garbage if it transmits first

### ISSUE THREAD

**Issue by diggit** [hathach/tinyusb#1039]
### Operating System

Linux

### Board

Custom board with ULPI PHY and STM32H730

### Firmware

cdc_dual example, with these changes:
- with only one CDC enabled
- loopback code disabled
- Instead of looping back data from host back to host, device transmits data periodically since boot
- CDC buffers raised to 4 kiB

```cpp
#include <inttypes.h>

const uint32_t coreFrequency = 16000000; // has to be configured according to core frequency

volatile uint32_t ticks = 0; // ticks are volatile variable incremented every 1 ms (systick).
void SysTick_Handler() {
	ticks = ticks + 1;
}

/*------------- MAIN -------------*/
int main(void)
{

	SysTick_Config(coreFrequency/1000); // 1 kHz systick ticking
	NVIC_EnableIRQ(UsageFault_IRQn);

	uint32_t prev = ticks; // save last TX timestamp (at first, we don't do tx, but nobody cares)
	uint8_t data = 0; // data for TX
	uint8_t buffer[64]; // buffer to catch spuriously received data
	unsigned int offset = 0; // buffer writing offset

	board_init();
	tusb_init();

	for(;;) {
		tud_task();

		const auto now = ticks;
		if (now - prev > 200) {
			prev = now;

			tud_cdc_write(&data, 1); // TX one byte
			tud_cdc_write_flush();
			data++; // increment data value for each transmission
		}

		if (tud_cdc_available()) {
			// capture all received data
			// assert(offset < sizeof(buffer) - 1);
			offset += tud_cdc_read(&buffer[offset], sizeof(buffer) - offset);
			// __BKPT(0);
		}
	}
	return 0;
}
```

### What happened ?

1) power on device and let i live (1s or so, just to have some data in TX buffer)
2) open UART (`ACM0`) in serial terminal (iserterm in ma case)
3) data in TX buffer are received by host, but also some new data ends in RX buffer (ERR led lit up and some garbage is in `buffer`, but I no (explicit) data transmission from host occurred

### How to reproduce ?

Basically replicate "What happened" scenario. I am not sure if this is issue of Linux or tinyUSB stack.

### Debug Log
[tusb_log.txt](https://github.com/hathach/tinyusb/files/7027747/tusb_log.txt)

**Comment by HiFiPhile** [comment]
You could try to capture USB traffic with Wireshark.

From the log there are OUT transfers came from PC, or the USB transfer complete interrupt was falsely triggered.

**Comment by hathach** [comment]
Please put long log as attached txt file to make issue easier to read

**Comment by hathach** [comment]
```C
pins::LedErr.setLevel(true);
```

it is not buildable, please also give the link to your fork that contain the exact code, it must be as minimal changes as possible.

**Comment by diggit** [comment]
I've updated code snippet. Now it should be C compatible and also not dependent on the rest of my code.
Just use `examples/device/cdc_dual_ports/src/main.c` example and replace main function block with that snippet.
Adjust `coreFrequency` according to your MCU frequency.

**Comment by hathach** [comment]
Please attached your example folder here, that makes sure we are testing the same piece of code.
